lib/testScenario/runtimeEnvManager.ts

```diff
@@ -0,0 +1,134 @@
+    if (eventEmitter && eventEmitter.summary) {
+      const environment = eventEmitter.summary.environment.syncVariablesTo();
+      let envContainer = this.runtimeEnvCollection.get(itemName);
+      if (!envContainer) {
+        envContainer = {};
+      }
+      switch (eventType) {
+        case "afterStep": {
+          envContainer.afterStep = environment;
+          break;
+        }
+        case "beforeStep": {
+          envContainer.beforeStep = environment;
+          break;
+        }
+      }
+      this.runtimeEnvCollection.set(itemName, envContainer);
+      const envFile = this.generateRuntimeEnvPath(itemName);
+      if (!existsSync(path.dirname(envFile))) {
+        mkdirpSync(path.dirname(envFile));
+      }
+      writeFileSync(this.generateRuntimeEnvPath(itemName), JSON.stringify(envContainer, null, 2));
+    }
+  };
+  public loadEnv = (fromStep: string) => {
+    const runtimeEnvPath = this.generateRuntimeEnvPath(fromStep);
+    if (!existsSync(runtimeEnvPath)) {
+      throw new Error(
+        `the last runtime env file ${runtimeEnvPath} for step '${fromStep}' did not exist. `
+      );
+    }
+    const runtimeEnvContainer = JSON.parse(
+      readFileSync(runtimeEnvPath).toString()
+    ) as RuntimeEnvContainer;
+
+    if (!runtimeEnvContainer.beforeStep) {
+      throw new Error(
+        `could not load last runtime env for step '${fromStep}', please check the file ${runtimeEnvPath} `
+      );
+    }
+    return runtimeEnvContainer.beforeStep;
+  };
+
+  public repopulateCollectionItems = (from?: string, to?: string) => {
+    if (!from && !to) {
+      return;
+    }
+    const collection = this.collection;
+    const fromIndex = from ? this.getStepIndex(collection, from) : 0;
+    let toIndex = to ? this.getStepIndex(collection, to) : collection.items.count() - 1;
+
+    if (fromIndex > toIndex) {
+      throw new Error(
+        `the step '${from}' is after the step '${to}', please check the command arguments.`
+      );
+    }
+
+    // keep the poller and delay steps in the last step.
+    while (to && toIndex + 1 < collection.items.count()) {
+      const itemName = collection.items.idx(toIndex + 1).name;
+      if (itemName.startsWith(`[generated]${to}`)) {
+        toIndex++;
+      } else {
+        break;
+      }
+    }
+    const items = [];
+    for (let idx = fromIndex; idx <= toIndex; idx++) {
+      if (idx < collection.items.count()) items.push(collection.items.idx(idx));
+    }
+    collection.items.repopulate(items);
+  };
+
+  public clean = () => {
+    if (this.hasCleanedFlag) {
+      return;
+    }
+    const fromStep = this.opts.from || this.opts.to ? this.collection.items.idx(0).name : undefined;
+    if (fromStep) {
+      this.cleanEnv(this.collection, fromStep);
+    }
+    this.hasCleanedFlag = true;
+  };
+
+  private generateRuntimeEnvPath = (step: string) => {
+    const envFilePostfix = "_env.json";
+    return path.join(this.runtimeFolder, ".runtime", step + envFilePostfix);
```

Comment:
The env path is not this folder. It should be in `generated/<RP>/<testScenario>/<runId>/runtime/`. All generated file should be together

Reply:
'this.runtimeFolder' is  same with the current report path ,  following  `generated/<RP>/api-version/<testScenarioFileName>/<runId>/<testScenarioName>` ,